docs/technical-reference/api/moduleTemplate-cr.md

```diff
@@ -49,6 +49,8 @@ If not specified, the **namespace** of the resource mentioned in **.spec.data**
 
 ### **.spec.customStateCheck**
 
+> **CAUTION:** This field is deprecated and will be removed in the future. 
```

Comment:
```suggestion
> **CAUTION:** This field was deprecated at the end of July 2024 and will be removed on {DATE}. 
```
We could also provide the deletion date. Is there any action required from the users?